### PR TITLE
Make Vega a peer dependency

### PR DESCRIPTION
**Pull request by domoritz**
This way users will know which version of vega we expect but we don't force anyone to install Vega (you can ignore the warning).

### CODE REVIEW

**Comment by jheer** [comment]
Sounds reasonable to me.

**Comment by domoritz** [comment]
Note that I also changed Vega-Embed 4 to use peer dependencies instead of depending directly on Vega or Vega-Lite.

**Comment by domoritz** [comment]
@tafsiri mentioned that it would be good to specify compatible versions with peer dependencies.

**Comment by domoritz** [comment]
@kanitw @arvind I would like to merge this. Any concerns?

File: package.json [hunk 1]
```diff
@@ -117,6 +117,9 @@
     "vega-util": "^1.9.0",
     "yargs": "^13.2.2"
   },
+  "peerDependencies": {
+    "vega": "^5.1.0"
```
Comment (kanitw) [inline package.json]: This version of Vega-Lite doesn't work with 5.2?  If so, remove ^?

Comment (domoritz) [reply package.json]: Good point.

**Comment by domoritz** [comment]
I switched to ~ for all dependencies. I think that's a good idea.

**Comment by domoritz** [comment]
We're pretty good about making new releases and updating dependencies so that using ~ instead of ^ will lead to fewer headaches.

**Pull request merged**